src/pages/home/index.js

```diff
@@ -2,18 +2,138 @@ import useTranslation from 'hooks/useTranslation';
 import { useLogoutMutation } from 'services/auth/auth';
 import Button from 'components/common/button';
 import MapView from 'components/map';
+import { useCreateTargetMutation, useGetTargetsQuery } from 'services/target/target';
+import { useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { z } from 'zod';
+import Input from 'components/form/input';
+import Select from 'components/form/select/select';
+import 'components/sideBar/styles.scss';
+import { useEffect, useState } from 'react';
+import { useTopicsQuery } from 'services/topics/topics';
+import SideBar from 'components/sideBar';
+import sideBarIcon from 'assets/aim-mark.png';
 
 import './styles.scss';
 
 const Home = () => {
   const t = useTranslation();
-  const [logout, { isLoading }] = useLogoutMutation();
 
   const handleLogout = () => logout().then(() => localStorage.removeItem('user'));
 
+  const [logout, { isLoading }] = useLogoutMutation();
+  const [createTarget, { isLoadingCreateTarget, isSuccess, error }] = useCreateTargetMutation();
+  const { data: topics } = useTopicsQuery();
+  const { data: targets } = useGetTargetsQuery();
+  const [topicsList, setTopicsList] = useState([]);
+  const [targetsList, setTargetsList] = useState([]);
+  const [latLng, setLatLng] = useState({});
+  const [currentPosition, setCurrentPosition] = useState({
+    ready: false,
+    where: [],
+    error: null,
+  });
+  const geoOptions = {
+    enableHighAccuracy: true,
+    timeout: 5000,
+    maximumAge: 0,
+  };
+  const geolocationEnabled = () => {
+    setCurrentPosition({ ready: false, error: null });
+    return navigator.geolocation.getCurrentPosition(
+      position => {
+        geoSuccess(position);
+      },
+      () => {},
+      geoOptions
+    );
+  };
+
+  const geoSuccess = position => {
+    setCurrentPosition({
+      ready: true,
+      where: [position.coords.latitude, position.coords.longitude],
+    });
+  };
+
+  useEffect(() => {
+    return geolocationEnabled();
+  }, []);
+
+  const parseTopics = () => {
+    let topicsParsedList = [];
+    topics?.topics.map(topic => {
+      topicsParsedList.push({
+        id: topic.topic.id,
+        label: topic.topic.label,
+      });
+    });
+    setTopicsList(topicsParsedList);
+  };
+
+  const parseTargets = () => {
+    let targetsParsedList = [];
+    targets?.targets.map(target => {
+      targetsParsedList.push([target.target.lat, target.target.lng, target.target.title]);
+    });
+    setTargetsList(targetsParsedList);
+  };
+
+  useEffect(() => {
+    parseTopics();
+  }, [topics]);
+  useEffect(() => {
+    parseTargets();
+  }, [targets]);
+  const schema = z.object({
+    radius: z.string().min(1),
+    title: z.string().min(1),
+    topic_id: z.string().min(1),
+  });
```

Comment:
Maybe you can extract this schema definition to another file, WDYT? Or declare it outside the component definition maybe?

Reply:
new schema file created